Sources/AWSSDKSwiftCore/Signers/V4.swift

```diff
@@ -40,21 +42,22 @@ extension Signers {
             return sha256(data).hexdigest()
         }
 
-        public init(credential: CredentialProvider, region: Region, service: String) {
+        public init(credential: CredentialProvider, region: Region, service: String, endpoint: String? = nil) {
             self.credential = credential
             self.region = region
             self.service = service
+            self.endpoint = endpoint
         }
 
         public func signedURL(url: URL, date: Date = Date(), expires: Int = 86400) -> URL {
             let datetime = V4.timestamp(date)
             let headers = ["Host": url.hostWithPort!]
             let bodyDigest = hexEncodedBodyHash(Data())
-            let credentialForSignature = getCredential()
+            resolveCredentialIfNeeded()
 
```

Comment:
Hey @noppoMan! welcome back!

This looks great. 

The reason for assigning the signer credential to a `credentialForSignature` constant here is due to the fact that a concurrent request could change the Signer credential at the moment when credentials expire and rotate on an instance / container setup. So I think we should preserve getCredential returning the current CredentialProvider if that makes sense?